### PR TITLE
Otel configuration updated to use expansion

### PR DESCRIPTION
**Pull request by BrynCooke**
Fixes https://github.com/apollographql/router/issues/1772
File and env access in configuration now use the generic expansion mechanism introduced in [#1759](https://github.com/apollographql/router/issues/1759).

```yaml
      grpc:
        key:
          file: "foo.txt"
        ca:
          file: "bar.txt"
        cert:
          file: "baz.txt"
```

Becomes:
```yaml
      grpc:
        key: "${file.foo.txt}"
        ca: "${file.bar.txt}"
        cert: "${file.baz.txt}"
```
or
```yaml
      grpc:
        key: "${env.FOO}"
        ca: "${env.BAR}"
        cert: "${env.BAZ}"
```

### CODE REVIEW

File: docs/source/configuration/overview.mdx [hunk 1]
```diff
@@ -316,7 +316,7 @@ See [Tracing in the Apollo Router](./tracing/).
 ### Automatic persisted queries (APQ)
 
 Automatic Persisted Queries (APQ) enable GraphQL clients to send a server the _hash_ of their query string, _instead of_ the query string itself. This can significantly reduce network usage for very large query strings.
-
+****
```
Comment (o0Ignition0o) [inline docs/source/configuration/overview.mdx:319]: is this line intentional ? is this some kinda `<hr>` magic ?

**Review by o0Ignition0o — APPROVED** [review]
Love the approach, expanding at the values level makes the yml much tidyer!

**Comment by abernix** [comment]
@BrynCooke Don't forget to "Fixes https://github.com/apollographql/router/issues/1772" in your PR body!

File: docs/source/configuration/overview.mdx (hunk 1 above)
Comment (abernix) [reply docs/source/configuration/overview.mdx:319]: hah, it might work.  i've only used `----` to do the same.

Comment (BrynCooke) [inline docs/source/configuration/overview.mdx:319]:
```suggestion
****
```

Comment (BrynCooke) [reply docs/source/configuration/overview.mdx]: Fixed in [a51e392](https://github.com/apollographql/router/pull/1774/commits/a51e3925ac55f78f8bcec1f43c0e42766469078a)

File: apollo-router/src/configuration/mod.rs [hunk 2]
```diff
@@ -36,50 +38,90 @@ use tower_http::cors::{self};
...
+struct Expansion {
+    prefix: Option<String>,
+    supported_modes: Vec<String>,
+}
 
-    // APOLLO_ROUTER_CONFIG_ENV_PREFIX is undocumented and unsupported currently. If there is demand from the community then it can be promoted to a top level feature.
-    Prefixed(String),
+impl Expansion {
+    fn default() -> Result<Self, ConfigurationError> {
+        // APOLLO_ROUTER_CONFIG_SUPPORTED_MODES and APOLLO_ROUTER_CONFIG_SUPPORTED_MODES are unspported and may change in future.
```
[the remark this answers is not in the document]
Comment (bnjjj) [reply apollo-router/src/configuration/mod.rs:49]: cc @BrynCooke

**Pull request merged**